tests/integration/cairo1_hint_locals/deploy/deploy_test.cairo

```diff
@@ -0,0 +1,114 @@
+
+    assert(prepare_result.contract_address != 0, 'prepared contract_address != 0');
+    assert(prepare_result.class_hash != 0, 'prepared class_hash != 0');
+    // TODO: check the length of the array, this produces: error: Variable was previously moved
+    // https://github.com/software-mansion/protostar/issues/1717
+    // assert(prepare_result.constructor_calldata.len() == 2_u32, 'constructor_calldata size == 2');
+
+    let prepared_contract = PreparedContract {
+        contract_address: prepare_result.contract_address,
+        class_hash: prepare_result.class_hash,
+        constructor_calldata: prepare_result.constructor_calldata
+    };
+    let deployed_contract_address = deploy(prepared_contract).unwrap();
+    assert(deployed_contract_address != 0, 'deployed_contract_address != 0');
+}
+
+#[test]
+fn test_deploy_with_ctor_invalid_calldata() {
+    let mut constructor_calldata = ArrayTrait::new();
+    constructor_calldata.append(1);
+
+
+    let class_hash = declare('with_ctor').unwrap();
+    assert(class_hash != 0, 'declared class_hash != 0');
+
+    let prepare_result = prepare(class_hash, constructor_calldata).unwrap();
+
+    let prepared_contract = PreparedContract {
+        contract_address: prepare_result.contract_address,
+        class_hash: prepare_result.class_hash,
+        constructor_calldata: prepare_result.constructor_calldata
+    };
+    deploy(prepared_contract).unwrap();
+}
+
+#[test]
+fn test_deploy_with_ctor_panic() {
+    let mut constructor_calldata = ArrayTrait::new();
+    constructor_calldata.append(1);
+    constructor_calldata.append(2);
+
+
+    let class_hash = declare('with_ctor_panic').unwrap();
+    assert(class_hash != 0, 'declared class_hash != 0');
+
+    let prepare_result = prepare(class_hash, constructor_calldata).unwrap();
+
+    let prepared_contract = PreparedContract {
+        contract_address: prepare_result.contract_address,
+        class_hash: prepare_result.class_hash,
+        constructor_calldata: prepare_result.constructor_calldata
+    };
+    deploy(prepared_contract).unwrap();
+}
+
+#[test]
+fn test_deploy_with_ctor_panic_check_err_code() {
+    let mut constructor_calldata = ArrayTrait::new();
+    constructor_calldata.append(1);
+    constructor_calldata.append(2);
+
+
+    let class_hash = declare('with_ctor_panic').unwrap();
+    assert(class_hash != 0, 'declared class_hash != 0');
+
+    let prepare_result = prepare(class_hash, constructor_calldata).unwrap();
+
+    let prepared_contract = PreparedContract {
+        contract_address: prepare_result.contract_address,
+        class_hash: prepare_result.class_hash,
+        constructor_calldata: prepare_result.constructor_calldata
+    };
+    match deploy(prepared_contract) {
+        Result::Ok(_) => {
+            assert(false, 'panic');
+        },
+        Result::Err(err) => {
+            assert(err == 71040974298098041582785264739058207689790761971295899515275945048047502191, 'proper error thrown');
```

Comment:
Can't we just assert it like `assert(err == 'My error message', 'proper error not thrown');`?

Reply:
We can address this separately. The problem is, Cairo has this limitation of the maximum length of any string (31). If we receive a long error message we may not be able to recognize what it really was. https://github.com/software-mansion/protostar/pull/1716#discussion_r1159419205

Reply:
Understood, but unfortunately this limitation is here to stay. I'd prefer we just raise exceptions with messages fitting 31 characters. The whole point of this is so that the user can then handle/check them in the tests without having to somehow compute this hashes.

Reply:
Sure. Nevertheless, I think this goes beyond of the scope of this PR and itself is a problem that should be discussed and addressed separately, if you don't mind how about following up there: https://github.com/software-mansion/protostar/issues/1725